target/linux/ramips/dts/mt7620a_elecom_wrc-733ghbk.dts

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "elecom,wrc-733ghbk", "ralink,mt7620a-soc";
+	model = "ELECOM WRC-733GHBK";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "wrc-733ghbk:red:wps";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power: power {
+			label = "wrc-733ghbk:blue:power";
+			gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "wrc-733ghbk:blue:wlan2g";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		wlan5g {
+			label = "wrc-733ghbk:blue:wlan5g";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		router {
+			label = "rt";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+	};
+
+	rtl8367rb {
+		compatible = "realtek,rtl8367b";
+		gpio-sda = <&gpio0 22 GPIO_ACTIVE_HIGH>;
+		gpio-sck = <&gpio0 23 GPIO_ACTIVE_HIGH>;
```

Comment:
`mdio-bus` can be used instead of bit-banged GPIO. See d4ac26ec4920036fe15414e47a0f1a3df6e8b9fb for details

Reply:
Thank you for your information. I tried it but failed to detect the switch.

```
[   10.279404] rtl8367b rtl8367rb: using MDIO bus 'mdio'
[   10.290477] rtl8367b rtl8367rb: unknown chip num:0070 ver:0070, mode:0070
[   10.304012] rtl8367b rtl8367rb: chip detection failed, err=-19
```

changes:

```
diff --git a/target/linux/ramips/dts/mt7620a_elecom_wrc-733ghbk.dts b/target/linux/ramips/dts/mt7620a_elecom_wrc-733ghbk.dts
index 8a4934657c..fba5c74ad5 100644
--- a/target/linux/ramips/dts/mt7620a_elecom_wrc-733ghbk.dts
+++ b/target/linux/ramips/dts/mt7620a_elecom_wrc-733ghbk.dts
@@ -68,9 +68,8 @@

        rtl8367rb {
                compatible = "realtek,rtl8367b";
-               gpio-sda = <&gpio0 22 GPIO_ACTIVE_HIGH>;
-               gpio-sck = <&gpio0 23 GPIO_ACTIVE_HIGH>;
                realtek,extif1 = <1 0 1 1 1 1 1 1 2>;
+               mii-bus = <&mdio0>;
        };
 };

@@ -129,13 +128,17 @@
 &ethernet {
        status = "okay";
        pinctrl-names = "default";
-       pinctrl-0 = <&rgmii1_pins>;
+       pinctrl-0 = <&rgmii1_pins &mdio_pins>;

        port@5 {
                status = "okay";
                mediatek,fixed-link = <1000 1 1 1>;
                phy-mode = "rgmii";
        };
+
+       mdio0: mdio-bus {
+               status = "okay";
+       };
 };

 &gpio0 {
@@ -150,7 +153,7 @@
 &pinctrl {
        state_default: pinctrl0 {
                gpio {
-                       ralink,group = "i2c", "uartf", "mdio";
+                       ralink,group = "i2c", "uartf";
                        ralink,function = "gpio";
                };
        };
```

Reply:
`0070` means `phy_id` is incorrect. See #2174 for details, you might have to pick this commit `kernel: rtl8366_smi: explicitly set phy addr for switch`

Reply:
What the number of phy_id shoud be used?

Reply:
In my case it's 29

Reply:
I see, hmm...In WRC-733GHBK, ```29``` didn't worked.

```
[   10.183414] rtl8367b rtl8367rb: switch phy addr=29
[   10.193022] rtl8367b rtl8367rb: using MDIO bus 'mdio'
[   10.204021] rtl8367b rtl8367rb: unknown chip num:ffff ver:ffff, mode:ffff
[   10.217555] rtl8367b rtl8367rb: chip detection failed, err=-19
```